mmdet/models/detectors/base.py

```diff
@@ -391,3 +399,8 @@ def show_result(self,
             warnings.warn('show==False and out_file is not specified, only '
                           'result image will be returned')
             return img
+
+
+    def export(self, img, img_metas, **kwargs):
```

Comment:
I'm sorry, but this won't work if we use NNCF.
The reason is as follows:
1. When we wrap a model by NNCF, model is wrapped by the class `NNCFNetwork` (it stores the original model as one of its field). 
    Note that `NNCFNetwork` has a transparent `__getattr__` that passes almost all attributes to the original model, so all calls of the models's methods will work.
2. But to make an adequate export of NNCF-compressed model, the function `torch.onnx.export ` **must** be called with the wrapped `NNCFNetwork` instead of the original model
3. But if we use this method `export`, the method is bound to the original model, not an `NNCFNetwork`.
    As the result, `self` in this method is the original model
    As the result, export of NNCF-compressed model will not work

This was the reason why we removed the method `export` from the model class.
Please, do not return the method `export` to the original class.
It will break our work.

Reply:
I made it work. If I use torch.onnx.eport I need to use it three time since my model is split on 3 parts (it is bad actually, but I have no idea how deal with it now). I would not like to make export.py script aware of export detail of particular model.

Moreover I have to hide details for mo.py conversion as well.

Reply:
@Ilya-Krylov -- oh, I see the reason.

I agree that nobody wants to make export.py script aware of export detail of particular model.

But, I'm sorry, this your change won't work:
```python
 model.export = getattr(model, 'export')
```

A simple example that you can run in ipython:
```python
class A:
    def f(self):
        print(type(self))

class B:
    def __init__(self, a):
        self._a = a
    def __getattr__(self, name):
        print(f"B.__getattr__({name})")
        return getattr(self._a, name)

a = A()
b = B(a)

b.f()       # will print 
            # B.__getattr__(f)
            # <class '__main__.A'>
b.f = getattr(b, "f")    # will print `B.__getattr__(f)`
b.f()                  # will print <class '__main__.A'>
b.f = b.f.__get__(b, type(b))  # will not help either
b.f()                 # will print <class '__main__.A'>

print(b.f)
# will print
# <bound method A.f of <__main__.A object at 0x7fabd91cf7d0>>
```
-- but I have an idea how to fix the issue, see my next comment.

Reply:
@LeonidBeynenson I don't like any variant. Because of my model split, because of `getattr` in NNCFNetwork, I implemented in 3rd way, though I don't like it as well :(

Reply:
Ilya, I'm sorry, but using this way is (IMHO) really dangerous. 
It is the path to the dark side, where the Spaghetti code waits all of us.

Please, note that as far as I can see my approach will allow to enable export of NNCF-compressed models  for your case, even if we have two models -- I looked at the code, and, since it may be converted to onnx, it will be converted to onnx after NNCF compression too.

Reply:
@Ilya-Krylov and @LeonidBeynenson , I'm sorry, and please don't laugh, but does not the following trick resolve the issue?

```python
class A:
    def f(self):
        print(type(self))

class B:
    def __init__(self, a):
        self._a = a

    def __getattr__(self, name):
        print(self.__dict__)
        print(f"B.__getattr__({name})")
        return getattr(self._a, name)

a = A()
b = B(a)

print('Before:')
b.f()
b.f = type(a).__dict__['f'].__get__(b)
print('\nAfter:')
b.f()
```

So, going back to export of the wrapped model, this way only the wrapper should be modified a bit to re-bind the method from the actual model class to the wrapper.

Reply:
Pavel, you are absolutely right.

I forgot that class gives the unbound method.

So, we can make the following change, which most proably will work
```python
export_method = type(model).export
compression_ctrl, model = create_compressed_model(model,
                                                      nncf_config,
                                                      dummy_forward_fn=dummy_forward,
                                                      resuming_state_dict=resuming_state_dict)
model.export = export_method.__get__(model)
```

Reply:
thanks, fixed